sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobAudience.java

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.models;
+
+/**
+ * The audience to be used when requesting a token from Azure Active Directory (AAD).
+ * Note: This audience only has an effect when authenticating a TokenCredential.
+ */
+public class BlobAudience implements Comparable<BlobAudience> {
```

Comment:
We have an existing pattern that we use for setting audience in other libraries. Can we use the same pattern?

- [Container Registry](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/ContainerRegistryAudience.java)
- [Cognitive Search](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/SearchAudience.java)
- [Personalizer](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/personalizer/azure-ai-personalizer/src/main/java/com/azure/ai/personalizer/models/PersonalizerAudience.java)